### PR TITLE
Feature - Created List molecule and List.Item atom components

### PR DESCRIPTION
**Pull request by MoiyadKayda**
## Screenshots:

Screen Shot 2021-02-10 at 17 55 39
Screen Shot 2021-02-10 at 17 55 46

## Checklist:

- [x] I have updated the CHANGELOG with a summary of my changes
- [x] I have updated the documentation accordingly

### CODE REVIEW

File: src/molecules/list/component.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+import { getClassName } from '@kibalabs/core';
+import { IMultiChildProps, OptionalProppedElement } from '@kibalabs/core-react';
+import styled from 'styled-components';
+
+import { IListItemProps, IListItemTheme, ListItem } from '../../atoms/listItem';
+import { defaultMoleculeProps, IMoleculeProps } from '../moleculeProps';
+
+export interface IListTheme {
+  ListItemTheme: IListItemTheme;
```
Comment (krishan711) [inline src/molecules/list/component.tsx:11]:
```suggestion
  listItemTheme: IListItemTheme;
```

File: src/atoms/listItem/component.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+export interface IListItemProps extends IComponentProps<IListItemTheme>, ISingleAnyChildProps {
+  itemKey: string;
+  isDisabled?: boolean;
+  isSelected?: boolean;
+  onClicked?(itemKey: string): void;
+}
+
+export const ListItem = (props: IListItemProps): React.ReactElement => {
+  const onClicked = !props.onClicked ? undefined : (): void => {
+    // this is to remove the type check error
```
Comment (krishan711) [inline src/atoms/listItem/component.tsx:73]: which error?

Comment (krishan711) [reply src/atoms/listItem/component.tsx:73]: the line below is definitely wrong, it should just be `props.onClicked(props.itemKey)`

Comment (MoiyadKayda) [reply src/atoms/listItem/component.tsx:73]: there is a type check error
Cannot invoke an object which is possibly 'undefined'.

**Review by krishan711 — APPROVED** [review]

**Pull request merged**